src/app-service.ts

```diff
@@ -228,11 +229,20 @@ export class AppService extends EventEmitter {
         try {
             await possiblePromise;
             res.send({});
-        } catch (e) {
-            res.send({
-                errcode: "M_UNKNOWN",
-                error: e instanceof Error ? e.message : ""
-            });
+        } catch (e: any) {
+            if (isMatrixHttpError(e)) {
```

Comment:
Would it be possible to document how to throw errors in https://github.com/matrix-org/matrix-appservice-node#example, as otherwise the information is likely to get lost!

Reply:
That make sense. I'll add it to the PR.

Reply:
Updated PR with suggested changes. Please have a look.